# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/blender/hotfix-sparc.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- blender-4.5.4/source/blender/blenlib/BLI_build_config.h.vanilla	2025-11-15 12:02:12.612847966 +0200
+++ blender-4.5.4/source/blender/blenlib/BLI_build_config.h	2025-11-15 12:04:34.168878509 +0200
@@ -357,6 +357,16 @@
 #  else
 #    define ARCH_CPU_BIG_ENDIAN 1
 #  endif
+#elif defined(__sparc__)
+#  define ARCH_CPU_SPARC_FAMILY 1
+#  define ARCH_CPU_BIG_ENDIAN 1
+#  if defined(__sparcv9)
+#    define ARCH_CPU_SPARC64 1
+#    define ARCH_CPU_64_BITS 1
+#  else
+#    define ARCH_CPU_SPARC 1
+#    define ARCH_CPU_32_BITS 1
+#  endif
 #elif defined(__loongarch_lp64)
 #  define ARCH_CPU_LOONG_FAMILY 1
 #  define ARCH_CPU_LOONG64 1
